### DISCUSSION TITLE
ColourizedFormatter cannot work after packaging with pyinstaller on Windows

### DISCUSSION THREAD

**Discussion by ilikeorange2006** [Kludex/uvicorn#2007] [Potential Issue]
After packaging with pyinstaller, `sys.stderr` will be `None`, and at line 47 of `uvicorn.logging`, it called sys.stdout.isatty() directly. So it will always raise an `AttributeError` like:
![image](https://github.com/encode/uvicorn/assets/7089412/e58495e7-f7bf-4d14-86c9-eecbefa732a8)
```
Traceback (most recent call last):
  File "logging\config.py", line 543, in configure
  File "logging\config.py", line 655, in configure_formatter
  File "logging\config.py", line 474, in configure_custom
  File "uvicorn\logging.py", line 47, in __init__
AttributeError: 'NoneType' object has no attribute 'isatty'
```
Of cause I can construct a custom logger config without `ColourizedFormatter`, but I think the default logger should avoid this error :)

**Comment by Kludex** [comment]
Do you have an MRE?

**Reply by ilikeorange2006** [reply]
> Do you have an MRE?

Uh yeah, the code is unnecessary, but I can show you:

```
import fastapi
import uvicorn

app = fastapi.FastAPI()


@app.get('/')
def anything():
    return 'Hello World'


uvicorn.run(app)
```

Then packaging it by pyinstaller:

```
pyinstaller test.py -w
```

Then run the dist exe.

```
dist\test\test.exe
```

Also, sys.stdin is also None, so line 37 of `uvicorn._subprocess` will also raise `AttributeError`. All of these should be checked.

**Reply by Kludex** [reply]
Does this PR solve your issue? https://github.com/encode/uvicorn/pull/1947/

**Reply by ilikeorange2006** [reply]
> Does this PR solve your issue? #1947

Half. Should also check `sys.stderr` at line 47 of `uvicorn.logging`
